File: android/NotifyBridge/app/src/main/java/com/alpware/notifybridge/notification/NotificationPayload.kt
package com.alpware.notifybridge.notification

/**
 * Represents a normalized Android notification before encryption and network transfer.
 */
data class NotificationPayload(
    val protocolVersion: Int = 2,
    val historyId: String = java.util.UUID.randomUUID().toString(),
    val eventType: String = "notification_upsert",
    val deviceId: String? = null,
    val packageName: String,
    val appName: String?,
    val title: String?,
    val text: String?,
    val postTime: Long,
    val contentHidden: Boolean = false,
    val appIconBase64: String? = null,
    val deviceName: String? = null,
    val notificationKey: String? = null,
    val canDismiss: Boolean = true,
    val canOpenOnPhone: Boolean = true,
    val canReply: Boolean = false,
    val replyAction: NotificationReplyActionPayload? = null
)

File: macos/NotifyBridgeMac/NotifyBridgeMac/NotifyBridgeMac/NotificationPayload.swift
//
//  NotificationPayload.swift
//  NotifyBridgeMac
//
//  Created by Mahmut Alperen Ünal on 13.05.2026.
//

import Foundation

/// Represents a decrypted notification received from the paired Android device.
struct NotificationPayload: Codable {
    let packageName: String
    let appName: String?
    let title: String?
    let text: String?
    let postTime: Int64
    let contentHidden: Bool?
    let appIconBase64: String?
    let deviceName: String?
    let notificationKey: String?
    let canDismiss: Bool?
    let canOpenOnPhone: Bool?
    let canReply: Bool?
    let replyAction: NotificationReplyActionPayload?
}

struct NotificationReplyActionPayload: Codable {
    let actionIndex: Int
    let label: String?
    let resultKey: String
}
